src/libcollections/string.rs

```diff
@@ -276,6 +277,7 @@ impl String {
     /// v[4] = 0xD800;
     /// assert!(String::from_utf16(v).is_err());
     /// ```
+    #[inline]
```

Comment:
this is easier to swallow than `fn from_utf8_lossy`, but again, not as trivial as the other methods marked in this patch